gdsfactory/component_reference.py

```diff
@@ -536,14 +529,6 @@ def connect(
             )
         )
 
-        if angle not in (0, 90, 180, 270):
-            origin_snapped = np.round(self.origin, 3)
-            shift = origin_snapped - self.origin
-            if any(shift):
-                self.origin = origin_snapped
-                for e in new.polygons:
-                    e.translate(*-shift)
```

Comment:
@amccaugh, this is what i used to do (as discussed in https://github.com/amccaugh/phidl/issues/162)... round the origin of the reference, then counter-shift the polygons by the sub-grid fraction that was rounded off the reference's origin. without accounting for other things like rotations, this should fix snapping issues between references